src/test/tests.ts

```diff
@@ -809,15 +805,16 @@ describe('Server', function () {
 
     setTimeout(() => {
       expect(connectionCallbackSpy.calledOnce).to.be.true;
-      expect(connectionCallbackSpy.getCall(0).args[0]).to.equal('Error');
+      expect(connectionCallbackSpy.getCall(0).args[0].message).to.equal('Error');
```

Comment:
Is this your idea?

```javascript
private sendError(connectionContext: ConnectionContext, opId: string, errorPayload: any,
                           overrideDefaultErrorType?: string): void {
    const sanitizedOverrideDefaultErrorType = overrideDefaultErrorType || MessageTypes.GQL_ERROR;
    if ([
        MessageTypes.GQL_CONNECTION_ERROR,
        MessageTypes.GQL_ERROR,
      ].indexOf(sanitizedOverrideDefaultErrorType) === -1) {
      throw new Error('overrideDefaultErrorType should be one of the allowed error messages' +
        ' GQL_CONNECTION_ERROR or GQL_ERROR');
    }

    let errorPayloadToSend = errorPayload;
    if (connectionContext.isLegacy) {
      errorPayloadToSend = errorPayload.message;
    }

    this.sendMessage(
      connectionContext,
      opId,
      sanitizedOverrideDefaultErrorType,
      errorPayloadToSend,
    );
  }
```

However in order to make this working I've to simulate a legacy client :( Or at least be able to change the isLegacy on the server for the test purposes